release/check-image-vulnerabilities/check-image-vulnerabilities.sh

```diff
@@ -0,0 +1,129 @@
+#
+set -euo pipefail
+
+IMAGE="${1:-}"
+VERSION="${2:-}"
+SUMMARY_PREFIX="${3:-}"
+
+check_not_empty \
+    IMAGE \
+    VERSION \
+    SUMMARY_PREFIX
+
+function main() {
+  local image="$1"
+  local version="$2"
+  local summary_prefix="$3"
+  local result_path="scan-result.json"
+
+  # Scan the image for vulnerabilities.
+  scan_image "$image" "$version" "$result_path"
+
+  # Count the number of vulnerabilities and fixable vulnerabilities for each severity.
+  # Use associative arrays to store counts by severity.
+  # Arrays are used via nameref parameters in print_vulnerability_status function.
+  # shellcheck disable=SC2034
+  declare -A vuln_counts
+  # shellcheck disable=SC2034
+  declare -A fixable_counts
+
+  # shellcheck disable=SC2034
+  for severity in CRITICAL IMPORTANT MODERATE; do
+    vuln_counts[$severity]="$(count_vulnerabilities "$severity" "$result_path")"
+    fixable_counts[$severity]="$(count_fixable_vulnerabilities "$severity" "$result_path")"
+  done
+
+  # Print the summary of the vulnerabilities.
+  gh_summary "### $summary_prefix $image:$version"
+  print_vulnerability_status vuln_counts fixable_counts
+
+  # Print the vulnerabilities table in a collapsible section.
+  # For the table to render correctly, we need to add a newline after the summary.
+  gh_summary "<details><summary>Click to expand details</summary>\n"
+  gh_summary "$(print_vulnerabilities_table "$result_path")"
+  gh_summary "</details>"
+
+  # If the failure flag is produced by print_vulnerability_status,
+  # exit with a failure status.
+  if [[ -f failure_flag ]]; then
+    exit 1
+  fi
+}
+
+# Scans the image for vulnerabilities.
+function scan_image() {
+  local image="$1"
+  local version="$2"
+  local result_path="$3"
+  roxctl image scan --output=json --force \
+    --severity="MODERATE,IMPORTANT,CRITICAL" \
+    --image="quay.io/${image}:${version}" > "$result_path"
+  gh_output result-path "$result_path"
+}
+
+# Counts the number of vulnerabilities for a given severity.
+function count_vulnerabilities() {
+  local severity="$1"
+  local result_path="$2"
+  jq ".result.summary.$severity" "$result_path"
+}
+
+# Counts the number of fixable vulnerabilities for a given severity.
+function count_fixable_vulnerabilities() {
+  local severity="$1"
+  local result_path="$2"
+  jq "[.result.vulnerabilities // [] | .[] | select(.cveSeverity == \"$severity\" and .componentFixedVersion != \"\")] | length" "$result_path"
+}
+
+# Prints the vulnerability status and an overview table of the vulnerabilities counts.
+function print_vulnerability_status() {
+  local -n vuln_counts_ref=$1
+  local -n fixable_counts_ref=$2
+
+  if (( fixable_counts_ref[CRITICAL] > 0 || fixable_counts_ref[IMPORTANT] > 0 )); then
+    local message="Found fixable critical or important vulnerabilities. See the step summary for details."
+
+    gh_log "error" "$message"
+    touch failure_flag
+    gh_summary "Status: ❌"
+    gh_summary "> $message"
```

Comment:
This is weird.

<img width="727" height="364" alt="image" src="https://github.com/user-attachments/assets/5feff477-ca43-47f4-96b0-808c391d172a" />

I **am** looking at the summary but this message makes me doubt myself. Is there some different summary I should look at?

> Found fixable critical or important vulnerabilities.

part is ok here

> See the step summary for details.

is not